pep-9999.rst

```diff
@@ -0,0 +1,187 @@
+
+
+Specification
+=============
+
+Size impact
+-----------
+
+To calculate the impact of these optimization efforts on size of the Python standard library we'll be using the `python3-libs`_ RPM package in Fedora. The Python standard library is vast and we believe it is not representative of what "batteries-included Python" means to users in modern constrained environments, such as containers. The `python3-libs`_ package in Fedora (and similarly in CentOS/RHEL), on the other hand, has already made a lot of choices on what should and should not be included in such an environment. On top of this, the `python3-libs` has already achieved a slightly smaller footprint by hardlinking ``.pyc`` bytecode files between optimization levels where they are identical.
+
+`python3-libs`_ in Fedora contains the following optional components: TODO XXX
+
+On the other hand, the following components are not included in `python3-libs`_ and are separated into different optional RPM packages: ``IDLE``, ``tkinter``, ``turtle``, ``turtledemo``, and ``test``. Also not included in `python3-libs`_ are the ``test`` subdirectories of the following components: ``ctypes``, ``distutils``, ``lib2to3``, ``sqlite3``, and ``unittest``.
+
+Size is calculated using ``du -c $UNPACKED/usr/lib64/python3.9`` (converted to MiBs by dividing by 1024 and rounding to 2 decimal places) on x86_64, where ``$UNPACKED`` is the unpacked python3-libs RPM.
+
+
+=======================================================================================  ==================================  =================  ===============
+  Option                                                                                 Size of ``/usr/lib(64)/python3.9``  Difference in MiB  Difference in %
+=======================================================================================  ==================================  =================  ===============
+Status quo before change                                                                 31.76 MiB
+Option 1: Shipping only ``.py`` files, and disabling creation of ``.pyc`` files          15.23 MiB                           -16.53 MiB         -52.05%
+Option 2: Shipping the non-optimized ``.pyc`` files and compressed ``.py`` source files  17.18 MiB                           -14.58 MiB         -45.91%
+=======================================================================================  ==================================  =================  ===============
+
+TODO XXX Option 1 in Miro's FC is 14.72 MiB	but I have it as 15.23 - why? Older version of python3.9? Or maybe ``encoding/*.pycs`` and ``pydoc_data/topics.pyc`` ? I kept them.
```

Comment:
@hroncok Btw, do you have any idea why this might be the case?

Reply:
I don't know. How do you measure?

Reply:
See text above:

    Size is calculated using ``du -c $UNPACKED/usr/lib64/python3.9`` (converted to MiBs by dividing by 1024 and rounding to 2 decimal places) on x86_64, where ``$UNPACKED`` is the unpacked python3-libs RPM.

I tried the measurement also in mock, and it was identical.

Reply:
Weird. No idea really, sorry. I've done this in mock.